src/traces/violin/index.js

```diff
@@ -0,0 +1,40 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+module.exports = {
+    attributes: require('./attributes'),
+    layoutAttributes: require('./layout_attributes'),
+    supplyDefaults: require('./defaults'),
+    supplyLayoutDefaults: require('./layout_defaults'),
+    calc: require('./calc'),
+    setPositions: require('../box/set_positions'),
+    plot: require('./plot'),
+    style: require('./style'),
+    hoverPoints: require('../box/hover'),
```

Comment:
Should violin show hover labels about the kernel density curve?

Reply:
It's tricky to figure out what someone would want to see labeled on the curve. The only thing I can think of that might be cool is a label that moves continuously (along the distribution axis) with the mouse, so you can look at a peak or a valley or something and see exactly what data value it's at... would help you read quantitative differences off several violins. Would a label like that get any value reported for the density? It wouldn't mean much on its own so could be omitted, though it would have meaning relative to other such values on the same or different violins.

Reply:
Here's what I came up with:

![peek 2017-10-27 14-36](https://user-images.githubusercontent.com/6675409/32119774-4cf8501a-bb24-11e7-896d-33d5f21951d1.gif)

Reply:
In my excitement at how beautiful this effect is, I missed an important piece: you should see both the kde and the y value in that label, so you can use it to read out the exact peak/valley locations for example. Should show just enough digits that each pixel is a different y value.

Reply:
Implemented in https://github.com/plotly/plotly.js/pull/2116/commits/bc6bc025ce3717c3e640f43eb71dd3ab2f719e01

Reply:
... now looks like:

![peek 2017-10-30 23-38](https://user-images.githubusercontent.com/6675409/32206450-7dc5adb0-bdcb-11e7-87b7-d1b9c323d223.gif)